### DISCUSSION TITLE
Cleanup files and orphans - python scripts

### DISCUSSION THREAD

**Discussion by m8dhouse** [open-webui/open-webui#5416] [Show and tell]
First version of 2 python scripts to cleanup /uploads folder and chat/files tables.

[Link to Github scripts](https://github.com/m8dhouse/openwebui-scripts/)

**cleanup.py** -> removes old chats older than 30d - and also the linked files
**orphans.py** -> removes orphan files (user deleted chat or documents)

Use at own risk. I just quickly created these and roughly tested it on a staging server. Seemed to be OK, but needs a bit more verification. Ideally you schedule the scripts daily in a cron job:

example:
0 3 * * * /usr/openweb/venv/bin/python3 /usr/openweb/cleanup.py --test N >> /usr/openweb/cleanlog/cleanup_cron.log 2>&1
15 3 * * * /usr/openweb/venv/bin/python3 /usr/openweb/orphan.py --test N >> /usr/openweb/orphanlog/cleanup_cron.log 2>&1

Not sure why open WebUI doesn't clean this already (I use PIP install) - now files remain on the server that could prove security risks.

**Comment by gaby** [comment]
@tjbck Thoughts on adding this as a feature to Open-WebUI. When organizations are using the UI they want to be able to auto-prune old chats.

Related to https://github.com/open-webui/open-webui/discussions/4580
